### ISSUE TITLE
Allow configuring CAN interfaces via LEA/CLEO

### ISSUE THREAD

**Issue by mbfm** [eclipse-opendut/opendut#128]
A user should be able to define a network interface as either Ethernet or CAN.  
For a CAN interface, further parameters should be provided.

Tasks:
- [x] Adjust opendut-types as illustrated below.
- [x] CLEO: Allow for configuring the type of an interface
- [x] LEA: Allow for configuring the type of an interface
- [x] EDGAR: Differentiate between Ethernet and CAN interfaces according to the interface type specified
- [x] CLEO: Allow for configuring CAN parameters
- [x] LEA: Allow for configuring CAN parameters
- [x] EDGAR: Configure CAN interfaces according to parameters set in their configurations

```rs
#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
pub struct PeerNetworkInterface {
    name: NetworkInterfaceName,
    configuration: PeerNetworkInterfaceConfiguration,
}

#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
pub enum PeerNetworkInterfaceConfiguration {
    Ethernet,
    Can {
        bitrate: i32,
        sample_point: f32,
        time_quanta: i32,
        propagation_segment: i8,
        phase_buffer_segment1: i8,
        phase_buffer_segment2: i8,
        synchronization_jump_width: i8,
        flexible_data_rate: bool,
        ...
    },
}
```

opendut-types/proto/opendut/types/peer/peer.proto:
```proto
message PeerNetworkInterface {
  opendut.types.util.NetworkInterfaceName name = 1;
  oneof configuration {
    EthernetInterfaceConfiguration ethernet = 11;
    CanInterfaceConfiguration can = 12;
  }
}
```

opendut-types/proto/opendut/types/util/net.proto:
```proto
message EthernetInterfaceConfiguration {}

message CanInterfaceConfiguration {
  int32 bitrate = 1;
  float sample_point = 2;
  int32 time_quanta = 3;
  int32 propagation_segment = 4;
  int32 phase_buffer_segment1 = 5;
  int32 phase_buffer_segment2 = 6;
  int32 synchronization_jump_width = 7;
  bool flexible_data_rate = 8;
  ...
}
```

Initially reported by @hafklin.

**Comment by hafklin** [comment]
The current state of this is that one can configure the interface type via CLEO as either CAN or Ethernet, but no selection of CAN parameters is possible.

**Comment by mirenz1** [comment]
Sufficient CAN configuration for LEA was implemented in #235 
Configuiration options in CLEO will be added with  [#276](https://github.com/eclipse-opendut/opendut/issues/276)
